examples/mqtt-client/main.c

```diff
@@ -40,10 +40,11 @@ static void fn(struct mg_connection *c, int ev, void *ev_data, void *fn_data) {
     struct mg_str subt = mg_str(s_sub_topic);
     struct mg_str pubt = mg_str(s_pub_topic), data = mg_str("hello");
     MG_INFO(("%lu CONNECTED to %s", c->id, s_url));
-    mg_mqtt_sub(c, subt, s_qos);
+    struct mg_mqtt_opts sub_opts = {.topic = subt, .qos = s_qos};
+    mg_mqtt_sub(c, &sub_opts);
     MG_INFO(("%lu SUBSCRIBED to %.*s", c->id, (int) subt.len, subt.ptr));
-
-    mg_mqtt_pub(c, pubt, data, s_qos, false);
+    struct mg_mqtt_opts pub_opts = {.topic = pubt, .message = data, .qos = s_qos, .retain = false};
```

Comment:
Note that our tests use many compilers, including an ancient MS VC 98 compiler, that does not understand designated intiialisers. Also, some C++ compilers (e.g. Arduinos on Cortex M0) does not accept designated initialisers if they are specified out of order of declaration.

Bottom line: avoid designated initialisers in the library code and examples. Use old idiom with declaration, memset, and set:

```c
struct mg_mqtt_opts opts;
memset(&opts, 0, sizeof(opts));
opts.qos = s_qos;
...
```